test/TestProjects/MgmtKeyvault/tests/Generated/Mock/SubscriptionExtensionsTest.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Net;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.TestFramework;
+using MgmtKeyvault.Models;
+
+namespace MgmtKeyvault.Tests.Mock
+{
+    /// <summary> Test for Subscription. </summary>
+    public partial class SubscriptionExtensionsMockTests : MockTestBase
+    {
+        public SubscriptionExtensionsMockTests(bool isAsync) : base(isAsync, RecordedTestMode.Record)
+        {
+            ServicePointManager.ServerCertificateValidationCallback += (sender, cert, chain, sslPolicyErrors) => true;
+            Environment.SetEnvironmentVariable("RESOURCE_MANAGER_URL", $"https://localhost:8443");
+        }
+
+        [RecordedTest]
+        public async System.Threading.Tasks.Task GetVaults()
+        {
+            // Example: List vaults in the specified subscription
+            var subscriptionExtensions = GetArmClient().GetSubscription(new ResourceIdentifier("/subscriptions/00000000-0000-0000-0000-000000000000"));
+            int? top = 1;
+
+            await foreach (var _ in subscriptionExtensions.GetVaultsAsync(top))
+            {
+            }
+        }
+
+        [RecordedTest]
+        public async System.Threading.Tasks.Task GetDeletedVaults()
+        {
+            // Example: List deleted vaults in the specified subscription
+            var subscriptionExtensions = GetArmClient().GetSubscription(new ResourceIdentifier("/subscriptions/00000000-0000-0000-0000-000000000000"));
+
+            await foreach (var _ in subscriptionExtensions.GetDeletedVaultsAsync())
+            {
+            }
+        }
+
+        [RecordedTest]
+        public async System.Threading.Tasks.Task CheckNameAvailabilityVault()
+        {
+            // Example: Validate a vault name
+            var subscriptionExtensions = GetArmClient().GetSubscription(new ResourceIdentifier("/subscriptions/00000000-0000-0000-0000-000000000000"));
```

Comment:
Can we change this method to use `SubscriptionResource.CreateResourceIdentifier`?
```suggestion
            var subscriptionExtensions = GetArmClient().GetSubscription(SubscriptionResource.CreateResourceIdentifier("00000000-0000-0000-0000-000000000000"));
```